agent/engine/docker_task_engine.go

```diff
@@ -326,6 +337,14 @@ func (engine *DockerTaskEngine) sweepTask(task *api.Task) {
 			seelog.Errorf("Error removing container reference from image state: %v", err)
 		}
 	}
+
+	// Clean metadata directory for task
+	if engine.cfg.ContainerMetadataEnabled {
+		err := engine.metadataManager.Clean(task.Arn)
+		if err != nil {
+			seelog.Warnf("Clean task metadata failed for task %s: %v", task, err)
```

Comment:
This should be `task.Arn`. Also, can  you add unit tests for `sweepTask` and exercise this code path?

Reply:
fixed the task.Arn thing. and the existing happy path [test](https://github.com/aws/amazon-ecs-agent/pull/981/files/b208d502e4abee9890d82c5056d2e333638a77c5#diff-3aedbbecbf5a79074e46dafde46c1536R401) exercises that code path.